app/views/railway_carriages/_form.html.erb

```diff
@@ -0,0 +1,32 @@
+<%= form_with(model: @railway_carriage.becomes(RailwayCarriage), local: true) do |form| %>
+  <%= render 'common/errors', resource: @railway_carriage %>
+
+  <div class="field">
+    <%= form.label :number %>
+    <%= form.text_field :number %>
```

Comment:
for what do you need this field in your form if you use callback( before_validation) to set up number.